src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj

```diff
@@ -21,6 +21,10 @@
       <File Include="$(CoreCLRBuildIntegrationDir)*" TargetPath="build" />
       <File Include="$(CoreCLRILCompilerDir)netstandard\*" TargetPath="tools/netstandard" />
       <File Include="sdk\Sdk.props" TargetPath="Sdk" />
+      <File Include="$(MSBuildThisFileDirectory)\..\..\..\..\native\libs\System.Globalization.Native\*" TargetPath="native/src/libs/System.Globalization.Native"/>
```

Comment:
These sources should be only in the previous item-group so that they are only included in the RID-specific package (e.g. runtime.linux-x64.microsoft.dotnet.ilcompiler). We do not need these sources in the entrypoint package (e.g. microsoft.dotnet.ilcompiler) - they won't be ever used from the entrypoint package.

Reply:
>These sources should be only in the previous item-group

I tried that but the user build does not pick up runtime.linux-x64.microsoft.dotnet.ilcompiler as a package reference in the csproj.
I will try some more, if I am still blocked, I will update PR with the change that conditionalizes the parent CMakeLists.txt.

Reply:
>Where is the runtime.linux-x64.microsoft.dotnet.ilcompiler package not getting picked up for you?

* The same steps are able to build the user binary without error by using this:
`<PackageReference Include="Microsoft.Dotnet.ILCompiler" Version="8.0.0-dev" /> 
`
* If I remove the PackageReference line, I get the same error as below.

```
ubuntu@ip-10-0-0-223:~/hello$ cat hello.csproj
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <PublishAot>true</PublishAot>
  </PropertyGroup>

  <ItemGroup>
     <PackageReference Include="runtime.linux-arm64.Microsoft.DotNet.ILCompiler" Version="8.0.0-dev" />
  </ItemGroup>
</Project>

ubuntu@ip-10-0-0-223:~/hello$ cat build.sh 
\rm -rf bin obj
~/dotnet-sdk/dotnet publish -c Release -r linux-arm64 /p:PublishAot=true /p:StaticICULinking=true -bl --self-contained -v d 2>&1 > log
ubuntu@ip-10-0-0-223:~/hello$ sh build.sh 
ubuntu@ip-10-0-0-223:~/hello$ tail log

       "/home/ubuntu/hello/hello.csproj" (Restore target) (1) ->
       (Restore target) -> 
         /home/ubuntu/hello/hello.csproj : error NU1102: Unable to find package runtime.linux-arm64.Microsoft.DotNet.ILCompiler with version (= 7.0.0-rc.1.22423.16)
       /home/ubuntu/hello/hello.csproj : error NU1102:   - Found 1 version(s) in repositoryPath [ Nearest version: 8.0.0-dev ]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.15

ubuntu@ip-10-0-0-223:~/hello$ cat nuget.config
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
    <clear />
    <add key="repositoryPath" value="/home/ubuntu/runtime/artifacts/packages/Debug/Shipping"/>
  </packageSources>
</configuration>

```

Reply:
You have to use the explicit package reference for the `-dev` package that you built locally. it overrides the reference to the shipping package version that is hardcoded in SDK.

Reply:
>You have to use the explicit package reference for the -dev package that you built locally. it overrides the reference to the shipping package version that is hardcoded in SDK.

I have the explicit package reference as:
` <PackageReference Include="runtime.linux-arm64.Microsoft.DotNet.ILCompiler" Version="8.0.0-dev" />`

```
ubuntu@ip-10-0-0-223:~/hello$ cat hello.csproj
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <PublishAot>true</PublishAot>
  </PropertyGroup>

  <ItemGroup>
     <PackageReference Include="runtime.linux-arm64.Microsoft.DotNet.ILCompiler" Version="8.0.0-dev" />
  </ItemGroup>
</Project>

```

Reply:
Our docs typically reference `Microsoft.DotNet.ILCompiler` and let the build figure out the RID-specific name. Referencing the RID-specific package directly works too. Is there something that does not work for you if you just reference `Microsoft.DotNet.ILCompiler` ?

Either way, I still do not see the problem with moving this to the RID-specific ItemGroup.

Reply:
>Either way, I still do not see the problem with moving this to the RID-specific ItemGroup.

Yes, I moved the to the previous itemGroup and it worked fine, I will update the change